### PR TITLE
Work around the limited Sonatype API responses

### LINKED ISSUE #49: Output on projects with more than 20 dependencies is incomplete
I stumbled upon this while hacking on a new feature (more on that later...).

When a project has more than 20 dependencies, the output of the plugin is incomplete; it will only mention the first 20 dependencies. This is caused by the Sonatype API never returning more than 20 items, even when you (politely) ask for more.

I managed to fix this, will try submit a PR this week.

### PR DESCRIPTION
**Pull request by mthmulders**
The API never returns more than 20 items. The code now chunks queries to ask for at most 20 items per query.

Fixes #49.

### CODE REVIEW

File: src/test/java/com/giovds/DependenciesToQueryMapperTest.java [hunk 1]
```diff
@@ -25,15 +31,18 @@ void should_return_lucene_query_based_on_given_dependencies() {
     @Test
     void should_return_multiple_lucene_query_based_on_given_dependencies_when_exceeding_limit() {
```
Comment (Giovds) [inline src/test/java/com/giovds/DependenciesToQueryMapperTest.java:32]: The additions here are awesome 🔥

I was playing around with the `IntStream` and stumbled on a `StringIndexOutOfBoundsException` when the amount of dependencies is dividable by the `maxItemsPerQuery`. The final iteration results in an empty query String when trying to remove the `+OR+` at the end of the query. I actually think this happens in the currently implementation as well, but it just very unlikely to happen due to the more random nature of the query length 🙈

```java
@ParameterizedTest
@ValueSource(ints = {25, 30, 40, 59, 60})
void should_return_multiple_lucene_query_based_on_given_dependencies_when_exceeding_limit(final int numberOfDeps) {
    final var division = (int) Math.ceil((double) numberOfDeps / DependenciesToQueryMapper.maxItemsPerQuery);
    final var otherDeps = IntStream.rangeClosed(1, numberOfDeps)
            .mapToObj(i -> String.format("bar-artifact-%d", i))
            .map(artifactId -> createDependency("org.apache.maven", artifactId, "1.2.3"))
            .collect(Collectors.toSet());

    final List<String> actualQueries = DependenciesToQueryMapper.mapToQueries(otherDeps);

    assertThat(actualQueries).hasSize(division).allMatch(query -> query.endsWith(")"));
}
```

Comment (mthmulders) [reply src/test/java/com/giovds/DependenciesToQueryMapperTest.java]: Nice catch. I've updated the test, made the query batching more robust, and added a few more edge cases for the number of dependencies.
